codec/address/bech32_codec.go

```diff
@@ -61,3 +130,38 @@ func (bc Bech32Codec) BytesToString(bz []byte) (string, error) {
 
 	return text, nil
 }
+
+func (cbc cachedBech32Codec) BytesToString(bz []byte) (string, error) {
+	key := conv.UnsafeBytesToStr(bz)
+	cbc.mu.Lock()
+	defer cbc.mu.Unlock()
+
+	if addr, ok := cbc.cache.Get(key); ok {
```

Comment:
**Not sure if this is possible** but in case the bytes match an existing bech32 address, it can create some trouble. Adding key prefixes may solve this or not sharing the same cache.

Reply:
See first comment. Currently, there are three different caches, but adding key prefixes may be a solution if we prefer just one.